pkg/cloudprivateipconfig/cloudprivateipconfig.go

```diff
@@ -0,0 +1,40 @@
+package cloudprivateipconfig
+
+import (
+	"errors"
+	"net"
+	"strings"
+)
+
+type IPFamily string
+
+const (
+	// IP Address Families
+	IPv4 IPFamily = "ipv4"
+	IPv6 IPFamily = "ipv6"
+)
+
+// CloudPrivateIPConfigNameToIP converts the resource name to net.IP. Given a
+// limitation in the Kubernetes API server (see:
+// https://github.com/kubernetes/kubernetes/pull/100950)
+// CloudPrivateIPConfig.metadata.name cannot represent an IPv6 address. To
+// work-around this limitation it was decided that the network plugin creating
+// the CR will fully expand the IPv6 address and replace all colons with dots,
+// ex:
+//
+// The IPv6 address fc00:f853:ccd:e793::54 will be represented
+// as: fc00.f853.0ccd.e793.0000.0000.0000.0054
+//
+// We thus need to replace every fifth character's dot with a colon.
+func CloudPrivateIPConfigNameToIP(name string) (net.IP, IPFamily, error) {
```

Comment:
nit: cloudprivateipconfig.go:29:6: func name will be used as cloudprivateipconfig.CloudPrivateIPConfigNameToIP by other packages, and that stutters; consider calling this NameToIP

(that's what the linter gives me, feel free to ignore)